pandas/tests/series/methods/test_sort_values.py

```diff
@@ -206,6 +206,28 @@ def test_mergesort_decending_stability(self):
         expected = Series([3, 2, 1, 1], ["c", "b", "first", "second"])
         tm.assert_series_equal(result, expected)
 
+    def test_sort_values_validate_ascending(self):
+        # GH41634
+        ser = Series([23, 7, 21])
+        expected = np.sort(ser.values)[::-1]
+
+        msg = 'For argument "ascending" expected type bool, received type str.'
```

Comment:
would it not be better to separate the raises ValueError test and the functional tests?
Also, would it not be better to parametrize the functional tests as follows:
```
@pytest.mark.parametrize("ascending", [False, 0, 1, True])
def test_sort_values_validate_ascending(self, ascending):
    ...
    sorted_ser = ser.sort_values(ascending=ascending)
    ...
    if ascending:
        expected = expected[::-1]
    ...
```

Reply:
Thanks @attack68. I agree, code will look more clean and short. Pushed a change for this, please check.

Also, I was thinking, does it make more sense to parameterize `test_sort_values_validate_ascending_for_value_error` test function and move code from line 54 to 68 (code attached below) in this test?

```
msg = 'For argument "ascending" expected type bool, received type NoneType.'
        with pytest.raises(ValueError, match=msg):
            ts.sort_values(ascending=None)
        msg = r"Length of ascending \(0\) must be 1 for Series"
        with pytest.raises(ValueError, match=msg):
            ts.sort_values(ascending=[])
        msg = r"Length of ascending \(3\) must be 1 for Series"
        with pytest.raises(ValueError, match=msg):
            ts.sort_values(ascending=[1, 2, 3])
        msg = r"Length of ascending \(2\) must be 1 for Series"
        with pytest.raises(ValueError, match=msg):
            ts.sort_values(ascending=[False, False])
        msg = 'For argument "ascending" expected type bool, received type str.'
        with pytest.raises(ValueError, match=msg):
            ts.sort_values(ascending="foobar")
```